lib/router/index.js

```diff
@@ -226,12 +226,16 @@ Router.prototype.unbind = function(route) {
 
   // Remove route in internal router
   var newRoutes = [];
-  _.each(this._privateRouter.routes[route.method], function(expressRoute) {
-    if (expressRoute.path != route.path) {
+  // Logic taken from https://github.com/balderdashy/sails/pull/3235/files
```

Comment:
I think this parses routes in `config/routes.js` in different formats and pushes them onto the router stack based on that